src/reading.rs

```diff
@@ -575,6 +572,8 @@ pub trait RevRead<'a>: Read {
 	/// # Arguments
 	///
 	/// * `amount`: The number of bytes by which to revert the previous read.
+	/// This may be at most the number of bytes returned by the previous call
```

Comment:
The rev_read call might return one byte after another, the limit is 27 bytes. Or maybe make it 32 because that's a round number :).

Reply:
Implementors of `RevRead` only need to guarantee that `revert` works for at most the number of bytes provided in the previous `rev_read` call (which is also exactly what the `RevReader` allows). Unless I misunderstand the original code, the offset provided in `SeekNeeded` is `needed - fnd_buf.len()`, where `needed >= 0` and `fnd_buf.len() <= rd_buf.len() == rd_len`. So, `needed - fnd_buf.len() > -rd_buf.len()`, which means that the amount of reverted bytes is less than or equal to the amount read in the last read call. So I think this requirement should suffice.

Reply:
It may be a good idea to construct a test case using the `RevReader`, but honestly I have no idea about the OGG format, so I would have difficulties with that ...

Reply:
You are right, I was wrong, I rest my case. And yeah, a test case would be really nice. I need to spend some time to think about it.